src/server/pps/server/api_server.go

```diff
@@ -553,9 +548,13 @@ func (a *apiServer) shardModuli(ctx context.Context, inputs []*ppsclient.JobInpu
 			}
 		}
 
-		b, err := a.noEmptyShards(ctx, inputs[modulusIndex], shardModuli[modulusIndex]*2, repoToFromCommit)
-		if err != nil {
-			return nil, err
+		b := false
```

Comment:
Shouldn't `b` be `true` instead of false?  That way, if `runEmpty` is set, we will double the modulus until it exceeds the parallelism.  The way it is right now, I think parallelism will just be ignored and you will always have 1 pod.